File: editorial/chap2-5/prob2-5-4.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		// 入力
		int N = sc.nextInt();
		
		// 出力
		for (int i = 2; i <= N; i++) {
			if (isPrime(i) == true) {
				if (i >= 3) System.out.print(" ");
				System.out.print(i);
			}
		}
		System.out.println();
	}
	static boolean isPrime(long N) {
		// 2 以上の整数 N に対し、N が素数であれば true、素数でなければ false を返す関数
		for (long i = 2; i <= N - 1; i++) {
			if (N % i == 0) {
				return false;
			}
		}
		return true;
	}
}

File: editorial/chap2-5/prob2-5-4.cpp
#include <iostream>
using namespace std;

bool isprime(int x) {
	for (int i = 2; i <= x - 1; i++) {
		// x を i で割った余りが 0 のとき、x は i で割り切れる
		if (x % i == 0) return false;
	}
	return true;
}

int main() {
	// 入力
	int N;
	cin >> N;
	
	// 答えの出力（空白区切りで出力する場合）
	for (int i = 2; i <= N; i++) {
		if (isprime(i) == true) {
			if (i >= 3) cout << " ";
			cout << i;
		}
	}
	cout << endl;
	return 0;
}